src/wallet/use-cases/update/update.spec.ts

```diff
@@ -0,0 +1,139 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { NotFoundException } from '@nestjs/common';
+import { plainToClass } from 'class-transformer';
+
+import { AccountType } from '@/shared/constants';
+import { UpdateWalletDTO } from '@/wallet/dto';
+import { IWalletRepository } from '@/wallet/interfaces';
+import { UpdateWalletUseCase } from '@/wallet/use-cases';
+import { FindBankByIdUseCase } from '@/bank/use-cases';
+import { IBankRepository } from '@/bank/interfaces';
+import { Bank } from '@/bank/infra/entities';
+import { Wallet } from '@/wallet/infra/entities';
+
+describe('Update Wallet Use Case', () => {
+  let updateWalletUseCase: UpdateWalletUseCase;
+  let findBankByIdUseCase: FindBankByIdUseCase;
+  let walletRepository: IWalletRepository;
+  let findByIdMock: jest.SpyInstance;
+  let updateMock: jest.SpyInstance;
+  let findBankByIdMock: jest.SpyInstance;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        UpdateWalletUseCase,
+        FindBankByIdUseCase,
+        {
+          provide: IWalletRepository,
+          useFactory: () => ({
+            update: jest.fn(),
+            findById: jest.fn(),
+          }),
+        },
+        {
+          provide: IBankRepository,
+          useFactory: () => ({
+            findById: jest.fn(),
+          }),
+        },
+      ],
+    }).compile();
+
+    updateWalletUseCase = module.get<UpdateWalletUseCase>(UpdateWalletUseCase);
+    findBankByIdUseCase = module.get<FindBankByIdUseCase>(FindBankByIdUseCase);
+    walletRepository = module.get<IWalletRepository>(IWalletRepository);
+
+    findByIdMock = jest.spyOn(walletRepository, 'findById');
+    updateMock = jest.spyOn(walletRepository, 'update');
+    findBankByIdMock = jest.spyOn(findBankByIdUseCase, 'execute');
+
+    findBankByIdMock.mockResolvedValue(bank);
+    updateMock.mockResolvedValue(NotFoundException);
+  });
+
+  const validId = '4e8b5d94-6b16-4a42-b6d1-dc58b553d109';
+  const invalidId = 'invalid-id';
+
+  const updateData: UpdateWalletDTO = {
+    id: validId,
+    bank_id: validId,
+    account_type: AccountType.CHECKING_ACCOUNT,
+    description: 'Nova descrição',
+  };
+
+  const bank = {
+    id: validId,
+    name: 'anyBank',
+    logo_url: 'anyURL',
+  } as Bank;
+
+  describe('execute', () => {
+    it('should be able to update a wallet successfully', async () => {
+      const updatedWallet = plainToClass(Wallet, updateData);
```

Comment:
Acho que esse teste está com muitas responsabilidades. Seria melhor separar um teste para garantir que cada um dos useCases/repositories estão sendo chamados corretamente, colocando o nome descritivo
ex:

"should call WalletRepository.findById with correct params"

Reply:
Entendi o seu ponto, por ser um padrão de clean arch separar as responsabilidades dos testes seria a melhor abordagem. Conversei com o Kayke sobre isso e ele disse que devido ao tempo das sprints e entregas não seria viável trocar isso agora, pois teria que refatorar todos os outros testes.